pkg/apply/taskrunner/task.go

```diff
@@ -26,25 +27,20 @@ type Task interface {
 	Name() string
 	Action() event.ResourceAction
 	Identifiers() object.ObjMetadataSet
-	Start(taskContext *TaskContext)
-	ClearTimeout()
+	Start(*TaskContext)
+	StatusUpdate(*TaskContext, object.ObjMetadata)
+	Cancel(*TaskContext)
```

Comment:
Note: this Cancel func is really just a placeholder. It will probably be removed when we pipe context through the TaskContext to handle cancellation and timeouts.